scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,143 @@
+        # is the gender 'Male'. This pattern repeats for all columns
+        # To verify this assumption for specific census tables, please check the
+        # table shells at: https://www2.census.gov/programs-surveys/acs/tech_docs/table_shells/2019/
+        val_prop_col = part_list[-1]
+
+        for part in part_list:
+            # set the base for special cases like median, etc.
+            if not measurement_assigned and 'measurement' in self.features:
+                if part in self.features['measurement']:
+                    stat_var.update(self.features['measurement'][part])
+                    measurement_assigned = True
+
+        # set the default statVar definition
+        if not measurement_assigned and 'measurement' in self.features and '_DEFAULT' in self.features['measurement']:
+            stat_var.update(self.features['measurement']['_DEFAULT'])
+
+        # set the populationType attribute
+        if 'populationType' not in stat_var:
+          stat_var['populationType'] = self._get_population_type(part_list)
+
+        # associate pvs to stat_var
+        if 'pvs' in self.features:
+            #p = property, k = dictionary of substrings in column that points to a propertyValue
+            for p, k in self.features['pvs'].items():
+                for c, v in k.items():
+                    ## check if the current key of dict matches with any substring from the tokens of a column name
+                    for part in part_list:
+                        # check if column substring matches with a column token
+                        if c == part:
+                            stat_var[p] = v
+                            # add inferred properties from property 'p' to the stat_var
+                            if 'inferredSpec' in self.features and p in self.features['inferredSpec']:
+                                  stat_var.update(self.features['inferredSpec'][p])
+
+
+      
+        #TODO: If quantityRanges occurs as values in multiple properties, find the property that relates to the column
+        #Suggestion from Prashnath: using _DISAMBIGUATING_POP_TYPE for pvs (income and earnings -- in the test) 
+
+        #TODO: Add units and scalingFactor
+
+        # if the column is Margin of Error, update statType key
+        if 'Margin of Error' in part_list:
+            stat_var['statType'] = 'marginOfError'
+
+        ## add Universe PVs based on the populationType of StatVar
+        dependent_properties = None
+        if 'universePVs' in self.features:
+            for elem in self.features['universePVs']:
+                if stat_var['populationType'] == elem['populationType']:
+                    # check if all constraints of this populationType is in stat_var
+                    if (set(elem['constraintProperties']).issubset(
+                            set(list(stat_var.keys())))):
+                        dependent_properties = list(elem['dependentPVs'].keys())
```

Comment:
shouldn't the dependant PVs be added to the stat_var as well?

Reply:
dependentPVs are defined like any other pvs based on previous feedback, so the `stat_var` dictionary has the dependentPVs before it reaches this code block.

Reply:
what about pvs that have to be added without any column, like ravi's table without column having the terms 'civilian' or 'unintstitutionalized' for some years although that is what the table code stands for.